packages/verified-fetch/README.md

```diff
@@ -658,6 +658,198 @@ Known Errors that can be thrown:
 3. `TypeError` - If the options argument is passed and is malformed.
 4. `AbortError` - If the content request is aborted due to user aborting provided AbortSignal. Note that this is a `AbortError` from `@libp2p/interface` and not the standard `AbortError` from the Fetch API.
 
+## Pluggability and Extensibility
+
+Verified‑fetch can now be extended to alter how it handles requests by using plugins.
+Plugins are classes that extend the `BasePlugin` class and implement the `VerifiedFetchPlugin`
+interface. They are instantiated with `PluginOptions` when the `VerifiedFetch` class is created.
+
+Each plugin must implement two methods:
+
+- **`canHandle(context: PluginContext): boolean`**
+  Inspects the current `PluginContext` (which includes the CID, path, query, accept header, etc.)
+  and returns `true` if the plugin can operate on the current state of the request.
+
+- **`handle(context: PluginContext): Promise<Response | null>`**
+  Performs the plugin’s work. It may:
+  - **Return a final `Response`**: This stops the pipeline immediately.
+  - **Return `undefined`**: This indicates that the plugin has only partially processed the request
+    (for example, by performing path walking or decoding) and the pipeline should continue.
+  - **Throw a `PluginError`**: This logs a non-fatal error and continues the pipeline.
+  - **Throw a `PluginFatalError`**: This logs a fatal error and stops the pipeline immediately.
+
+Plugins are executed in a chain (a **plugin pipeline**):
+
+1. **Initialization:**
+   - The `VerifiedFetch` class is instantiated with a list of plugins.
+   - When a request is made via the `fetch` method, the resource and options are parsed to
+     create a mutable `PluginContext` object.
+
+2. **Pipeline Execution:**
+
+   - The pipeline repeatedly checks, up to a maximum number of passes (default = 3), which plugins
+     are currently able to handle the request by calling each plugin’s `canHandle()` method.
+   - Plugins that have not yet been called in the current run and return `true` for `canHandle()`
+     are invoked in sequence.
+     - If a plugin returns a final `Response` or throws a `PluginFatalError`, the pipeline immediately
+       stops and that response is returned.
+     - If a plugin returns `undefined`, it is assumed to have updated the context (for example, by
+       performing path walking), and the pipeline proceeds to a new pass where plugins are re‑evaluated.
+   - If no plugin modifies the context (i.e. no change to `context.modified`) and no final response is
+     produced after iterating through all plugins, the pipeline exits and a default “Not Supported”
+     response is returned.
```

Comment:
I would add here a note about execution order. 

- Plugin execution is determined by the order they are passed in. The loop iterates multiple times to ensure dependent plugins are executed even if you pass `plugins` in the wrong order of execution.